catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/RawUdpDataToMTSPacketDecoder.java

```diff
@@ -143,37 +147,41 @@ protected void decode(ChannelHandlerContext ctx, DatagramPacket msg, List<Object
         notNull(msg, "msg must be non-null");
         notNull(outputList, "outputList must be non-null");
 
-        checkSecuritySubject(msg);
+        LOCK.lock();
```

Comment:
Do we need to lock this entire block?  If the critical section is just writing to the buffer, can we just lock that line?